test/unit/test_exports/test_derived_quantities/test_derived_quantities.py

```diff
@@ -330,3 +337,23 @@ def test_filename_ends_with_csv():
     """Checks that an error is raised when filename doesn't end with .csv"""
     with pytest.raises(ValueError, match="filename must end with .csv"):
         DerivedQuantities(filename="coucou")
+
+
+def test_set_derived_quantitites_wrong_type():
+    """Checks an error is raised when festim.DerivedQuantities is set with the wrong type"""
+    flux1 = SurfaceFlux(field="solute", surface=1)
+
+    combinations = [flux1, "coucou", 1, True]
+
+    for dq_combination in combinations:
+        with pytest.raises(
+            TypeError,
+            match="festim.DerivedQuantities must be a list",
+        ):
+            DerivedQuantities(dq_combination)
+
+    with pytest.raises(
+        TypeError,
+        match="festim.DerivedQuantities must be a list of festim.DerivedQuantity",
+    ):
+        DerivedQuantities([flux1, 2])
```

Comment:
This tests that an error is raised at instantiation but what about after instantiation?

Reply:
Yes, it should be addressed if we want to introduce a check of item validaty